core/platform/email/mailgun_email_services_test.py

```diff
@@ -164,41 +191,51 @@ def test_batch_send_to_mailgun(self) -> None:
-                'plaintext_body 😂',
-                'Hi abc,<br> 😂')
-            self.assertTrue(resp)
-
-    def test_mailgun_key_or_domain_name_not_set_raises_exception(self) -> None:
+        with self.swap_api_key_secrets_return_secret, swap_urlopen_context:
+            with swap_request_context, swap_api, swap_domain:
+                resp = mailgun_email_services.send_email_to_recipients(
+                    'a@a.com',
+                    ['b@b.com', 'c@c.com', 'd@d.com'],
+                    'Hola 😂 - invitation to collaborate',
+                    'plaintext_body 😂',
+                    'Hi abc,<br> 😂')
+                self.assertTrue(resp)
+
+    def test_mailgun_key_not_set_raises_exception(self) -> None:
         """Test that exceptions are raised when API key or domain name are
         unset.
         """
         # Testing no mailgun api key.
         mailgun_exception = self.assertRaisesRegex(
             Exception, 'Mailgun API key is not available.')
-        with mailgun_exception:
-            mailgun_email_services.send_email_to_recipients(
-                'a@a.com',
-                ['b@b.com', 'c@c.com', 'd@d.com'],
-                'Hola 😂 - invitation to collaborate',
-                'plaintext_body 😂',
-                'Hi abc,<br> 😂')
+        with self.swap_api_key_secrets_return_none, mailgun_exception:
+            with self.capture_logging() as logs:
+                mailgun_email_services.send_email_to_recipients(
+                    'a@a.com',
+                    ['b@b.com', 'c@c.com', 'd@d.com'],
+                    'Hola 😂 - invitation to collaborate',
+                    'plaintext_body 😂',
+                    'Hi abc,<br> 😂')
+                self.assertIn(
+                    'Cloud Secret Manager is not working.\nNoneType: None', logs
+                )
 
+    def test_mailgun_domain_name_not_set_raises_exception(self) -> None:
         # Testing no mailgun domain name.
         swap_api = self.swap(feconf, 'MAILGUN_API_KEY', 'key')
         mailgun_exception = self.assertRaisesRegex(
             Exception, 'Mailgun domain name is not set.')
-        with swap_api, mailgun_exception:
-            mailgun_email_services.send_email_to_recipients(
-                'a@a.com',
-                ['b@b.com', 'c@c.com', 'd@d.com'],
-                'Hola 😂 - invitation to collaborate',
-                'plaintext_body 😂',
-                'Hi abc,<br> 😂')
+        with self.swap_api_key_secrets_return_none, swap_api, mailgun_exception:
+            with self.capture_logging() as logs:
+                mailgun_email_services.send_email_to_recipients(
+                    'a@a.com',
+                    ['b@b.com', 'c@c.com', 'd@d.com'],
+                    'Hola 😂 - invitation to collaborate',
+                    'plaintext_body 😂',
+                    'Hi abc,<br> 😂')
+                self.assertIn(
+                    'Cloud Secret Manager is not working.\nNoneType: None', logs
```

Comment:
I feel like the "\nNoneType: None' is not very understandable. Could it be made more human-understandable so that a developer who sees it knows what to do?

Reply:
Added an explanatory comment.